### DISCUSSION TITLE
Is there a clever way to activate and deactivate dynamically certain constraints when solving NLP?

### DISCUSSION THREAD

**Discussion by BIT-glimmer** [casadi/casadi#3235] [Q&A] [answered]
Hello everyone,

I am trying to use casadi to solve an online MPC problem? During the solution of the NLP, some algebraic constraints (nonlinear functions of decision variables)  should be activated and deactivated occasionally, depending on things like time or contact detection.
One of the way that I can think of is add all those potential constraints at first, and pass Inf and -Inf for the boundaries of those should be deactivated . The other way is to use if_else statement when constructing the symbolic constraints, and pass some indicators as parameters to activate or deactivate those constraints? Should these two methods work or there exists any better way?  And will these two methods cause numerical instability?

Because the codes I need to change are cubersome? I want a better understanding before a concrete try. Thanks a lot!

**Comment by BIT-glimmer** [comment]
I tried the first way, it works. However, it seems that the boundaries should be set as a reasonable interval instead of [-Inf, Inf], to overcome the numerical instability due to much too relaxed boundaries. Still open to any alternative solutions!

**Comment by jaeandersson** [comment, marked as answer]
Do not use "if_else", it makes the problem nonsmooth. Infinite bonds are fine. Typically, the solver will eliminate such bonds before solving the problem anyway.
